tests/test_utils.py

```diff
@@ -688,35 +694,39 @@ def test_simple(self):
 
         da = xr.open_dataset(self.nc_poslons).tas
         da["lon"] -= 360
-        yr_st = 2050
-        yr_ed = 2059
+        yr_st = "2050"
+        yr_ed = "2059"
 
         out = subset.subset_bbox(
-            da, lon_bnds=self.lon, lat_bnds=self.lat, start_yr=yr_st, end_yr=yr_ed
+            da, lon_bnds=self.lon, lat_bnds=self.lat, start_date=yr_st, end_date=yr_ed
         )
         assert np.all(out.lon >= np.min(self.lon))
         assert np.all(out.lon <= np.max(self.lon))
         assert np.all(out.lat >= np.min(self.lat))
         assert np.all(out.lat <= np.max(self.lat))
-        np.testing.assert_array_equal(out.time.dt.year.max(), yr_ed)
-        np.testing.assert_array_equal(out.time.dt.year.min(), yr_st)
+        np.testing.assert_array_equal(out.time.dt.year.max(), np.array(int(yr_ed)))
```

Comment:
Do we need an `np.array` here ?

Reply:
Unfortunately, yes. The format of `out.time.dt.year.max()` is a numpy integer while the `yr_st` and `yr_ed` are strings. Even after casting them to `int`, their datatype is not the same so loading them into `np.array` objects is necessary. The alternative would be to cast the values of `out.time.dt.year.max()` to `str` for comparison.

Reply:
But why did it work before ?

Reply:
yr_st and yr_ed were sent in as integers using the `start_year` and `end_year` keywords. In newer versions of xclim, we've moved away from this by demanding `str` objects that can be parsed as "YYYY", "YYYY-MM" or "YYYY-MM-DD" and passed to `start_date` or `end_date`. For backward compatibility, using the old keywords is allowed, but a warning is raised that this will be removed in a future version.

When we implemented these changes, we added a test to assert that a warning is raised, but we didn't modify existing tests using the old/soon-to-be-deprecated calls. This PR does that, hence why the changes here are needed.

Reply:
Ah, I see your question now. It's weird that simply passing them as `int` worked previously. For some reason, the tests weren't working when only casting `str` to `int`. I figured loading them into arrays made the most sense. Mysterious.

Reply:
Ok, no biggy, was just curious.